_posts/2019-07-29-custom-domains-function-ingress.md

```diff
@@ -0,0 +1,235 @@
+
+* Nginx IngressController
+
+Traefik and Skipper are also supported, but we'll use Nginx for this tutorial. Install Nginx with the following:
+
+```
+helm install stable/nginx-ingress --name nginxingress --set rbac.create=true
+```
+
+* cert-manager (optional)
+
+Using TLS is optional, but highly recommended, so I will show you how to set it up.
+
+Use the [1.0 SSL for Kubernetes](https://docs.openfaas.com/reference/ssl/kubernetes-with-cert-manager/#10-ssl-for-the-gateway) documentation to set up cert-manager and an issuer. You can skip the steps for creating a `Certificate` and `Ingress` entry for the gateway.
+
+### Deploy `IngressOperator`
+
+We'll now deploy the IngressOperator. I'll be [following instructions from the OpenFaaS documentation](https://docs.openfaas.com/reference/ssl/kubernetes-with-cert-manager/#20-ssl-and-custom-domains-for-functions).
+
+```sh
+git clone https://github.com/openfaas-incubator/ingress-operator
+cd ingress-operator
+
+kubectl apply -f ./artifacts/operator-crd.yaml
+kubectl apply -f ./artifacts/operator-rbac.yaml
+kubectl apply -f ./artifacts/operator-amd64.yaml
+```
+
+The above creates the CRD definition, RBAC roles and then deploys the Operator itself.
+
+### Create a static website
+
+We can use the Nginx static website template to create a function or microservice. Once deployed, we'll move on to creating its custom domain name entry and then a CRD entry for it.
+
+```sh
+export OPENFAAS_PREFIX=alexellis2
+
+faas template pull https://github.com/alexellis/static-site-template
+faas new --lang static-site-nginx my-homepage --prefix ${OPENFAAS_PREFIX}
+```
+
+Set `OPENFAAS_PREFIX` with your own Docker Hub username, or private registry address and repo.
+
+Create the homepage:
+
+```sh
+echo "<html>Hello world</html>" > ./my-homepage/index.html
+```
+
+The following will build a container image, push it to the remote registry and then deploy it to OpenFaaS, resulting in a static-website.
+
+```sh
+faas-cli up -f my-homepage.yml
+```
+
+Your traditional URL will appear on:
+
+```sh
+faas-cli describe -f my-homepage.yml my-homepage
+
+Name:                my-homepage
+Status:              Ready
+Replicas:            1
+Available replicas:  0
+Invocations:         0
+Image:               alexellis2/my-homepage:latest
+Function process:    
+URL:                 http://206.189.242.89:8080/function/my-homepage
+Async URL:           http://206.189.242.89:8080/async-function/my-homepage
+Labels:              faas_function : my-homepage
+```
+
+We'll now create a DNS entry and then the custom hostname mapping and TLS certificate along with that.
+
+### Create a DNS A record for your sub-domain
+
+My testing domain is `myfaas.club` and it cost me a very minimal amount of money from [namecheap.com](https://namecheap.com/). I can create sub-domains for each function, or for any testing that I do. If you are familiar with DNS-management, then create an A record for the IP address the LoadBalancer created by Nginx.
```

Comment:
*then create an A record for the IP address the LoadBalancer created by Nginx*, I think we are missing an `of` before `the LoadBalancer`